richtextfx/src/main/java/org/fxmisc/richtext/GenericStyledArea.java

```diff
@@ -406,42 +407,78 @@ protected void invalidated() {
      *                                                                        *
      * ********************************************************************** */
 
-    private final ObjectProperty<Consumer<MouseEvent>> onOutsideSelectionMousePress = new SimpleObjectProperty<>(e -> {
-        CharacterHit hit = hit(e.getX(), e.getY());
-        moveTo(hit.getInsertionIndex(), SelectionPolicy.CLEAR);
-    });
+    @Deprecated private final ObjectProperty<Consumer<MouseEvent>> onOutsideSelectionMousePress = new SimpleObjectProperty<>( null );
```

Comment:
Could you move all deprecated methods down to the bottom of the file? It gets them out of the way.

Reply:
Tried to do this, but then the compiler complains about an **Illegal forward reference** so I've reverted this back to the way it was.

Reply:
Tried to do this but then the compiler complains about **Illegal forward references** !?